specification/schemaregistry/data-plane/Microsoft.EventHub/preview/2018-01-01-preview/schemaregistry.json

```diff
@@ -29,104 +29,302 @@
+            "name": "X-Schema-Type",
+            "in": "header",
+            "type": "string",
+            "description": "Serialization type for the schema being registered.",
+            "required": true,
+            "enum": [
+              "avro"
+            ],
+            "x-ms-enum": {
+              "name": "SerializationType",
+              "modelAsString": true,
+              "values": [
+                {
+                  "value": "avro",
+                  "name": "AVRO",
+                  "description": "Avro Serialization schema type"
+                }
+              ]
+            },
+            "x-nullable": false
+          },
+          {
+            "name": "schema-content",
+            "in": "body",
+            "description": "String representation of the schema being registered.",
+            "required": true,
+            "schema": {
+              "type": "string"
+            }
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
           }
         ],
         "responses": {
           "200": {
-            "description": "Describe the result of a successful operation.",
+            "description": "OK",
+            "headers": {
+              "Location": {
+                "type": "string",
+                "description": "URL location of schema, identified by schema group, schema name, and version."
+              },
+              "X-Schema-Type": {
+                "type": "string",
+                "description": "Serialization type for the schema being registered."
+              },
+              "X-Schema-Id": {
+                "type": "string",
+                "description": "References specific schema in registry namespace."
+              },
+              "X-Schema-Id-Location": {
+                "type": "string",
+                "description": "URL location of schema, identified by schema ID."
+              },
+              "X-Schema-Version": {
+                "type": "integer",
+                "description": "Version of the returned schema."
+              }
+            },
             "schema": {
-              "$ref": "#/definitions/Result"
+              "$ref": "#/definitions/SchemaId"
             }
           },
           "default": {
-            "description": "Error response describing why the operation failed.",
+            "description": "An error response received from the Azure Schema Registry service.",
             "schema": {
-              "$ref": "#/definitions/ErrorResponse"
+              "$ref": "#/definitions/ServiceErrorResponse"
             }
           }
         }
       }
     }
   },
   "definitions": {
-    "Result": {
-      "description": "Sample result definition",
+    "SchemaId": {
+      "type": "object",
+      "description": "JSON Object received from the registry containing schema identifiers.",
       "properties": {
-        "sampleProperty": {
+        "id": {
           "type": "string",
```

Comment:
I think this should have "required": true. Autorest is generating it as being possibly undefined.

Reply:
Checked with @jhendrixMSFT on this one, Joel correct me if I'm wrong here.

So there is a field 'x-nullable' that is set to false by default.  Not all language generators have implemented support for it yet though, so some languages will have nullable fields.  Joel noted C# works, Java ended up always being nullable if I remember correctly, and obviously JS ends up being nullable as well.

Reply:
So, looks like there is no action item here for now because x-nullable defaults to false.  Any action item would be on the generator side.